### PR TITLE
add use_cache option

### PR DESCRIPTION
**Pull request by stillmatic**

### CODE REVIEW

**Comment by okayzed** [comment]
what's the reasoning behind it?

File: src/template.py [hunk 1]
```diff
@@ -256,8 +257,8 @@ def build_template(self, name, data):
 
     def setup_template(self, template):
 
-        if template in YAML_CACHE:
-            return self.setup_template_from_cache(template)
+        if template in YAML_CACHE and use_cache:
```
Comment (okayzed) [inline src/template.py]: i think `use_cache` will be undefined here, it should be `self.use_cache`

**Comment by stillmatic** [comment]
hey, thanks for quick review!

context: i am getting familiar with plaitpy's syntax by prototyping in a jupyter notebook. i'd like to be able to rerun one cell (`t = plaitpy.Template("data_template.yaml")`) and regenerate the record. without this option, i need to restart the kernel every time I make a change to `data_template.yaml`, which slows down iteration speed dramatically.

let me know if there's anything else you'd like to see in the PR, too.

**Comment by okayzed** [comment]
thanks for trying plaitpy out!

i think it might be more convenient to do: `plaitpy.template.YAML_CACHE.clear()` or `plaitpy.template.YAML_CACHE = {}` when you want to reset the template cache. 

this PR (as is) has a syntax error for me since use_cache is not referenced properly. if you want to go ahead with this PR, have to change the line i commented on to reference `self.use_cache` and then i will re-test it.

**Comment by stillmatic** [comment]
thanks! i've updated the branch to fix the syntax error, but your suggestion to clear the cache does work for me as well, so this PR isn't strictly needed.

**Comment by okayzed** [comment]
one other thing that can be done is bumping the version in src/version.py, then i will build and upload it to pip after merging

**Comment by okayzed** [comment]
closing for now. if still interested, i will merge

**Pull request closed without merging**